src/lib/number.jl

```diff
@@ -45,19 +46,57 @@ end
 
 @adjoint function sincos(x)
   s, c = sincos(x)
-  (s, c), ((s̄, c̄),) -> (s̄*c - c̄*s,)
+  (s, c), ((s̄, c̄),) -> (s̄ * c - c̄ * s,)
 end
 
-@adjoint a // b = (a // b, c̄ -> (c̄ * 1//b, - c̄ * a // b // b))
+@adjoint a // b = (a // b, c̄ -> (c̄ * 1 // b, -c̄ * a // b // b))
 
 @nograd floor, ceil, trunc, round, hash
 
 # Complex Numbers
 
-@adjoint (T::Type{<:Complex})(re, im) = T(re, im), c̄ -> (nothing, real(c̄), imag(c̄))
+@adjoint (T::Type{<:Complex})(re, im) =
+  T(re, im), c̄ -> (nothing, real(c̄), imag(c̄))
 
 @adjoint real(x::Number) = real(x), r̄ -> (real(r̄),)
 @adjoint conj(x::Number) = conj(x), r̄ -> (conj(r̄),)
-@adjoint imag(x::Number) = imag(x), ī -> (real(ī)*im,)
+@adjoint imag(x::Number) = imag(x), ī -> (real(ī) * im,)
+
+DiffRules._abs_deriv(x::Complex) = x / abs(x)
+
+ # Function add adjoint for Fastmath operations
+ # DiffRules is used to find derivatives for each operation
+function fastmath()
+
+  for f in keys(fast_op)
+    unarydx, binarydx = DiffRules.hasdiffrule(:Base, f, 1) ?
+                        DiffRules.diffrule(:Base, f, :x) : nothing,
+      DiffRules.hasdiffrule(:Base, f, 2) ?
+      DiffRules.diffrule(:Base, f, :x, :y) : nothing
+
+    fastf = fast_op[f]
+
+    if unarydx != nothing
```

Comment:
This seems like a strange way to write it. Why not just test `hasdiffrule` directly here?